pkg/kubelet/apis/podresources/server.go

```diff
@@ -79,7 +83,60 @@ func (p *podResourcesServer) List(ctx context.Context, req *v1alpha1.ListPodReso
 		podResources[i] = &pRes
 	}
 
-	return &v1alpha1.ListPodResourcesResponse{
+	return &podresourcesapi.ListPodResourcesResponse{
 		PodResources: podResources,
 	}, nil
 }
+
+type numaIDs []int64
+
+func (a numaIDs) Len() int           { return len(a) }
+func (a numaIDs) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a numaIDs) Less(i, j int) bool { return a[i] < a[j] }
+
+// GetAllocatableResources returns information about all the devices known by the server
+func (p *podResourcesServer) GetAllocatableResources(context.Context, *podresourcesapi.AllocatableResourcesRequest) (*podresourcesapi.AllocatableResourcesResponse, error) {
+	allDevices := p.devicesProvider.GetAllDevices()
+	var respDevs []*podresourcesapi.ContainerDevices
+
+	resourceNames := []string{}
+	for resourceName := range allDevices {
+		resourceNames = append(resourceNames, resourceName)
+	}
+	sort.Strings(resourceNames)
+
+	for _, resourceName := range resourceNames {
+		resourceDevs := allDevices[resourceName]
+
+		// group devices by NUMA node
+		numaNodes := []int64{}
+		NUMADeviceIDs := map[int64][]string{}
+		for devID, dev := range resourceDevs {
+			for _, node := range dev.GetTopology().GetNodes() {
+				numaNode := node.GetID()
+				NUMADeviceIDs[numaNode] = append(NUMADeviceIDs[numaNode], devID)
+				numaNodes = append(numaNodes, numaNode)
+			}
+		}
+		sort.Stable(numaIDs(numaNodes))
```

Comment:
why do we need a stable sort here, it has more swap operation than usual Sort?

Reply:
No real reason. Changing to `sort.Sort`.